rand_distr/src/poisson.rs

```diff
@@ -158,14 +159,15 @@ mod test {
         let mut sum_u64 = 0;
         let mut sum_f64 = 0.;
         for _ in 0..1000 {
-            let s_u64: u64 = poisson.sample(&mut rng);
+            let s_u64 = poisson.sample(&mut rng) as u64;
```

Comment:
This may trigger undefined behavior if the sampled float is larger than `u64::MAX` (see #795). I think this is fixed in some recent Rust version, but I'm not sure whether it's fixed in 1.36.

Reply:
Ah, the old impl was panicking in the case of overflow. If we are to remove the `Distribution<u64>`, then I think we can simply remove code which tests it.